_posts/2019-02-06-til-send-after.md

```diff
@@ -0,0 +1,114 @@
+---
+author: Sean Callan
+author_link: https://github.com/doomspork
+categories: til
+date: 2019-02-06
+layout: post
+title:  TIL how to use `Process.send_after/4` to create reocurring checks
+excerpt: >
+  Want to schedule something to run later? Need a reocurring task? Today we learn how!
+---
+
+Executing code later or creating reocurring tasks can be tricky but did you know we can accomplish this in Elixir with just a process?
+With a GenServer, `Process.send_after/4`, and the `handle_info/2` callback we have everything we need.
+
+Let's look at `Process.send_after/4` and the expected arguments:
+
+```elixir
+send_after(dest, msg, time, opts \\ [])
+```
+
+- The `dest` argument takes the `pid` or name of our process, we'll use a named GenServer for our example.
+- The `msg` we want sent to the process, this can be just about any data structure but we'll stick with a simple atom.
+- Provided as milliseconds, `time` is how long until we want our message sent.
+- Last but not least, options.
+
+That's all well and good but where exactly does the `msg` go after `time` has elapsed?
+Great question!
+
+The less often used `handle_info/2` callback is how these messages are handled.
+Just like `handle_cast/2`, `handle_info/2` takes two parmeters: the first will be our `msg` from above and second the current state.
+
+That's enough to get us going but if you're interested to learn more about `Process.send_after/4` be sure to check out the [official documentation](https://hexdocs.pm/elixir/Process.html#send_after/4).
+
+For the sake of demonstrating how to use our aforementioned tools to perform reocurring work let's build a simple module to output the current time every 30 seconds.
+Since we'll be working with a GenServer we can rely on `init/1` as a good place to kick off our the reocurring work using `Process.send_after/4` and a message of `:tick`:
+
+```elixir
+@ten_seconds 10000
```

Comment:
I think you mentioned "every 30 seconds" above. Let's keep it consistent.

Reply:
I originally went with 30 but it takes too long. I thought I replaced them all, good catch. Thanks for the review @SophieDeBenedetto! Hope you're having a great week 😁